packages/extension-ui/src/Popup/RestoreJson.tsx

```diff
@@ -47,28 +48,35 @@ function Upload ({ className }: Props): React.ReactElement {
 
   const _onChangeFile = useCallback(
     (file: Uint8Array): void => {
-      let json: KeyringPair$Json | undefined;
+      setAccountsInfo(() => []);
+
+      let json: KeyringPair$Json | KeyringPair$Json[] | undefined;
 
       try {
-        json = JSON.parse(u8aToString(file)) as KeyringPair$Json;
+        json = JSON.parse(u8aToString(file));
+        json = json && ([] as KeyringPair$Json[]).concat(json);
         setFile(json);
       } catch (e) {
         console.error(e);
         setFileError(true);
       }
 
-      json && jsonGetAccountInfo(json)
-        .then((accountInfo: ResponseJsonGetAccountInfo) => setAccountInfo(accountInfo))
-        .catch((e) => {
-          setFileError(true);
-          console.error(e);
+      if (Array.isArray(json)) {
+        json.forEach((pair) => {
+          jsonGetAccountInfo(pair)
+            .then((accountInfo) => setAccountsInfo((old) => [...old, accountInfo]))
+            .catch((e) => {
+              setFileError(true);
+              console.error(e);
+            });
         });
+      }
     }, []
   );
 
   const _onRestore = useCallback(
     (): void => {
-      if (!file || !password) {
+      if (!file) {
```

Comment:
TO BE DECIDED
password requirement is currently disabled.

Reply:
I have tested it. I think the updated behavior aligns with our plan.

In single wallet Recovery, user is required to type in the password of the wallet for validation. However, in batch Recovery typing in all passwords is not practical, so it's better to skip the validation step here and leave it to the signing process.